File: ur5_control_panel/ur5_control_panel/main_window.py
import sys

from PySide6.QtCore import Slot
from PySide6.QtWidgets import (
	QApplication,
	QMainWindow,
	QWidget,
	QVBoxLayout,
	QPushButton,
)
from pathlib import Path


class MainWindow(QMainWindow):
    def __init__(self) -> None:
        super().__init__()
        self.setWindowTitle("UR5 Control Panel")
        self.resize(800, 800)

        central = QWidget(self)
        layout_exit = QVBoxLayout(central)

        self.exit_button = QPushButton("Salir", central)
        self.exit_button.setObjectName("exit_button")
        self.exit_button.clicked.connect(self.on_exit_clicked)
        layout_exit.addWidget(self.exit_button)

        self.setCentralWidget(central)

    @Slot()
    def on_exit_clicked(self) -> None:
        app = QApplication.instance()
        if app is not None:
            app.quit()


def main() -> int:
    app = QApplication(sys.argv)
    apply_stylesheet(app)
    window = MainWindow()
    window.show()
    return app.exec()


def apply_stylesheet(app: QApplication) -> None:
    """Carga el stylesheet desde resource/style.qss del paquete."""
    try:
        # Ruta al archivo style.qss en la carpeta resource del paquete
        qss_path = Path(__file__).resolve().parent.parent / "resource" / "style.qss"
        if qss_path.exists():
            app.setStyleSheet(qss_path.read_text(encoding="utf-8"))
    except Exception as e:
        print(f"Warning: No se pudo cargar style.qss: {e}")


if __name__ == "__main__":
    sys.exit(main())

File: ur5_interfaz_panel/src/mainWindow.cpp
#include "ur5_interfaz_panel/mainWindow.hpp"
#include <QLabel>
#include <QFile>
#include <QDebug>
#include <QPushButton>
#include <QLineEdit>
#include <QHBoxLayout>
#include <QGridLayout>



MainWindow::MainWindow(QApplication *app, QWidget *parent)
    : QMainWindow(parent), app_(app) {
    
    // Crear widget central, contenedor de los elementos
    //DEclaraciones que luego se moveran a hpp
    QWidget *menuWidget_;
    QWidget *rvizWidget_;
    QHBoxLayout *mainLayout_;

    

    centralWidget_ = new QWidget();
    menuWidget_ = new QWidget();
    menuWidget_->setObjectName("menuWidget");
    menuWidget_->setFixedWidth(300);

    // Inicializar rvizWidget antes de usarlo
    rvizWidget_ = new QWidget();
    rvizWidget_->setObjectName("rvizWidget"); 

    mainLayout_ = new QHBoxLayout(centralWidget_);
    mainLayout_->addWidget(menuWidget_);
    mainLayout_->addWidget(rvizWidget_);

    Layout_menu_ = new QVBoxLayout(menuWidget_); //Layout_menu_ estará dentro de menuWidget_
    
    // Crear título
    QLabel *titleLabel = new QLabel("Interfaz UR5");
    titleLabel->setStyleSheet("font-size: 20px; font-weight: bold;");
    titleLabel->setAlignment(Qt::AlignCenter);
    Layout_menu_->addWidget(titleLabel);
    
    // Agregar espacio
    Layout_menu_->addStretch();
    
    // Crear botón de salir
    exitButton_ = new QPushButton("Salir");
    exitButton_->setObjectName("exitButton");
    exitButton_->setFixedSize(150, 40);
    Layout_menu_->addWidget(exitButton_, 0, Qt::AlignCenter);
    
    // Conectar señal del botón
    connect(exitButton_, &QPushButton::clicked, this, &MainWindow::onExitClicked);
    
    // Crear widget contenedor
    QWidget *containerWidget = new QWidget();
    containerWidget->setObjectName("buttonContainer");

    // Crear layout y asignarlo al widget
    QHBoxLayout *buttonLayout = new QHBoxLayout(containerWidget);
    buttonLayout->addWidget(new QPushButton("Botón 1"));
    buttonLayout->addWidget(new QPushButton("Botón 2"));

    // Agregar el widget (no el layout) al layout principal
    Layout_menu_->addWidget(containerWidget);

    // Crear grid para controles
    QGridLayout *controlGrid = new QGridLayout();
    controlGrid->addWidget(new QLabel("X:"), 0, 0);
    controlGrid->addWidget(new QLineEdit(), 0, 1);
    Layout_menu_->addLayout(controlGrid);
    
    // Configurar widget central
    setCentralWidget(centralWidget_);
    
    // Configurar ventana
    setWindowTitle("Panel de Control UR5");
    resize(400, 300);
    
    // Cargar estilos QSS
    loadStyleSheet();

}

void MainWindow::loadStyleSheet() {
    QFile styleFile(":/style/style.qss");
    if (styleFile.open(QFile::ReadOnly | QFile::Text)) {
        QString styleSheet = QLatin1String(styleFile.readAll());
        app_->setStyleSheet(styleSheet);
        qDebug() << "Estilos QSS cargados correctamente";
        styleFile.close();
    } else {
        qDebug() << "Error: No se pudo cargar el archivo style.qss";
    }
}

void MainWindow::onExitClicked() {
    app_->quit();
}

MainWindow::~MainWindow() {
}